### ISSUE TITLE
iOS 14+ photo gallery permission "selected photos" does not affect MessageInput (gallery preview)

### ISSUE THREAD

**Issue by Rongix** [GetStream/stream-chat-flutter#937]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS

### What version are you using?

stream_chat_flutter 3.4.0

### What happened?

iOS 14+ introduces Enhanced Privacy Experience in Photos Apps with a new "Limited Library". Stream chat example app's default chat input does not work correctly when picking the "selected photos" privacy option. Nothing shows in the gallery preview. 

**Expected**
For example, other chat clients/providers show the message: "You've given SuperChatApp access to a selected number of photos and videos." with a gallery of selected photos and a button to manage settings -> select more photos or change settings completely in the system settings

### Steps to reproduce

```bash
1. Open and run the example project "stream-chat-flutter/packages/stream_chat_flutter/example/lib/main.dart"
2. It should start on the chat, click on the icon with an image (first icon)
3. Dialog about permissions should show up
4. Select "Selected photos..." 
5. Pick a few photos
6. Selected photos do not show. Instead, there is a notification about no access to the gallery.
```

### Supporting info to reproduce

Use iOS 14+, preferably the newest iOS version available, I'm using iOS 15.2 currently and it is reproducible there.

Offending component: "MessageInput"

### Relevant log output

```shell
-
```

### Flutter analyze output

```shell
-
```

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 2.10.2, on macOS 12.2.1 21D62 darwin-x64, locale en-GB)
    • Flutter version 2.10.2 at /Users/pawelfranitza/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (11 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc4)
    • Android SDK at /Users/pawelfranitza/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0-rc4
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 55.1.5
    • Dart plugin version 211.6693.108

[✓] VS Code (version 1.64.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (2 available)
    • iPhone 13 Pro (mobile) • ----- • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome
      98.0.4758.109

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Rongix** [comment]
Video demonstrating the issue:

https://user-images.githubusercontent.com/52180711/156196420-d84bef27-cdaf-421d-970f-f4b8eeb7f48d.mov

**Comment by Rongix** [comment]
Hey, sorry, I can't. I've made a report quite some time ago and don't work on a project that uses StreamChat anymore :/ 
I've made an effort to check recent changes in the Changelog, and I don't see any changes related to permissions so I assume, it is not resolved.

**Comment by imtoori** [comment]
solved by #1092
